# diff_drive_hw.yaml
diffbot:
  ros__parameters:
    update_rate: 100  # Hz

    # Hardware
    hardware:
      plugin: "diff_drive_hw/DiffDriveSystem"
      parameters:
        wheel_radius: 0.05
        wheel_separation: 0.30
        max_wheel_speed: 20.0
        cmd_timeout: 0.5

    # Joints (names must match controllers & URDF)
    joints:
      - left_wheel_joint
      - right_wheel_joint

    # State & command interfaces per joint
    state_interfaces:
      - position
      - velocity
    command_interfaces:
      - velocity